### PR TITLE
Implement DetectCloakedMultiplier

### PR DESCRIPTION
**Pull request by MustaphaTR**
Similar story to #16707.

TESTCASE makes it so having Radar Dome or Allied Tech Center increase cloak detection range of Radar Jammer.

### CODE REVIEW

**Review by abcdefg30 — DISMISSED** [review]

**Comment by netnazgul** [comment]
Test case will be removed from actual PR, right? ;)

**Comment by abcdefg30** [comment]
Yes. This PR doesn't have a second 👍 yet though, so I would keep it for the next person testing it.

File: OpenRA.Mods.Common/Traits/DetectCloaked.cs [hunk 1]
```diff
@@ -24,9 +26,29 @@ public class DetectCloakedInfo : ConditionalTraitInfo
...
+		}
+
+		public WDist Range
+		{
+			get
+			{
+				if (IsTraitDisabled)
+					return WDist.Zero;
+
+				var revealsShroudModifier = rangeModifiers.Select(x => x.GetDetectCloakedModifier());
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/DetectCloaked.cs]: `revealsShroudModifier` is probably copy-paste, please change that to something like `detectCloakedModifier`.

**Comment by reaperrr** [comment]
👍 after that minor fix and test case removal.

**Comment by MustaphaTR** [comment]
Updated and removed the testcase.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/3b2f8a84684c7eeb8c48bbcc55bdf032a350075a...f3e83db7bd88d08dde9ea0a5557e3b3379a25611

File: OpenRA.Mods.Common/Traits/DetectCloaked.cs [hunk 2]
```diff
@@ -24,9 +26,29 @@ public class DetectCloakedInfo : ConditionalTraitInfo
...
 
-	public class DetectCloaked : ConditionalTrait<DetectCloakedInfo>
+	public class DetectCloaked : ConditionalTrait<DetectCloakedInfo>, INotifyCreated
 	{
+		IDetectCloakedModifier[] rangeModifiers;
+
 		public DetectCloaked(DetectCloakedInfo info)
 			: base(info) { }
+
+		void INotifyCreated.Created(Actor self)
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Traits/DetectCloaked.cs]: Doesn't this explicitly override `ConditionalTrait.INotifyCreated.Created` and should have been a `protected override Created` method starting with `base.Created` with no `INotifyCreated`  interface implementation within this file?

Comment (reaperrr) [reply OpenRA.Mods.Common/Traits/DetectCloaked.cs]: Good catch.
Should have been

```
		protected override void Created(Actor self)
		{
			rangeModifiers = self.TraitsImplementing<IDetectCloakedModifier>().ToArray();

			base.Created(self);
		}
```

Comment (pchote) [reply OpenRA.Mods.Common/Traits/DetectCloaked.cs]: This is also a problem for `AutoTarget`, `PrimaryBuilding`, `GrantConditionOnDeploy`, `Explodes`, `KillsSelf`, `ReloadAmmoPool`, `ActorSpawnManager`, `Chronoshiftable`, `ResourcePurifier`.

I am working on a lint check that will prevent this from happening again in the future.

**Pull request merged**